cmd/orderimports/orderimports.go

```diff
@@ -0,0 +1,207 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// verify that all the imports have our preferred order.
+// https://github.com/kubernetes/kubeadm/issues/2515
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"go/ast"
+	"go/parser"
+	"go/token"
+	"log"
+	"os"
+	"path/filepath"
+	"regexp"
+	"sort"
+	"strings"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+var (
+	includePath = flag.String("include-path", "", "only files with paths matching this regex are touched")
+	ignoreFile  = flag.String("ignore-file", "zz_generated", "files matching this regex are ignored")
```

Comment:
the support to make changes to the files in-place can be added via a `-w` flag,, similarly to `gofmt`:

```
gofmt -h
...
   -w    write result to (source) file instead of stdout
```

Reply:
I wrote a tool based on `gofmt` to do the formatting works:https://github.com/xinydev/fmtimports (prototype), and the second commit is generated by this tool,will make some changes to make it better understood, now there are some tricks(not good) for adding blank lines between imports

```sh
find ./cmd/kubeadm -name "*.go" | xargs fmtimports -w -r '^"[^.]*"$ OTHERS ^"((\w)*\.)*k8s\.io\/([^k]|k[^u]|ku[^b]|kub[^e]|kube[^r]).*\"$ ^\"k8s.io\/kubernetes.*\"$'
```

Reply:
ok, note that ideally cmd/orderimports/orderimports.go should be the tool that can both verify and update.